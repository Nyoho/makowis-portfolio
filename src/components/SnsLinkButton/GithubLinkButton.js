// @flow
import BaseButton from './BaseButton';
import { GithubLinkButton as className } from './GithubLinkButton.module.css';

const idToUrl = (id: string) => `https://github.com/${id}`;

const GithubLinkButton = ({ githubId }: { githubId: string }) =>
  BaseButton({ className, url: idToUrl(githubId), icon: 'github' });

export default GithubLinkButton;
