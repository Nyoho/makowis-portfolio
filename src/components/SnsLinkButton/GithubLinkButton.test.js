/* eslint-env jest */
// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import GithubLinkButton from './GithubLinkButton';

it('renders GithubLinkButton', () => {
  const tree = renderer
    .create(<GithubLinkButton url="https://github.com/makowis" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
