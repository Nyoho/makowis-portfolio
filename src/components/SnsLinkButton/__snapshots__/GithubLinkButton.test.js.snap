// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GithubLinkButton 1`] = `
<a
  className="SnsLinkButton"
  href="https://github.com/makowis"
  style={
    Object {
      "background": "black",
    }
  }
/>
`;
